contracts/utils/contracts/src/MixinReentrancyGuardRichErrors.sol

```diff
@@ -0,0 +1,46 @@
+/*
+
+  Copyright 2019 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+
+import "./RichErrors.sol";
+
+
+contract MixinReentrancyGuardRichErrors is
```

Comment:
When do we use the `Mixin` patterns for errors rather than the library pattern?

Reply:
We use the library pattern when the rich revert occurs in a library like LibBytes. We'll use the `Mixin` pattern for contracts if we decide to stick with it.